internal/persistence/queries.go

```diff
@@ -9,22 +9,31 @@ import (
 	"github.com/dhth/hours/internal/types"
 )
 
-func InsertNewTL(db *sql.DB, taskID int, beginTs time.Time) error {
-	stmt, err := db.Prepare(`
+var ErrCouldntRollBackTx = errors.New("couldn't roll back transaction")
+
+func InsertNewTL(db *sql.DB, taskID int, beginTs time.Time) (int64, error) {
```

Comment:
@ccoVeille Would be great to get your take on this.

Several functions that create new rows in the database are now returning the ID of the newly inserted row using the following:

```go
func (sql.Result) LastInsertId() (int64, error)
```

The rest of the codebase deals with IDs as `int`s. Now, I can either refactor the whole codebase to use `int64` for IDs, or I can convert `int64` to `int` in such functions before returning them. The latter could cause issues on 32 bit systems, but then again, it is not really a major target architecture.

Any recommendations for dealing with this?

Reply:
Convert the int64 to int.

Even if someone try to use on a 32bits architecture, reaching the 2 billions as an identifier would be incredible with your tool. Then sql.LastInsertID would never return a number greater than this, as the arch cannot create such integer. So converting the int64 to int will be OK

Reply:
Yeah, good point 👍

Reply:
I think no one will build your tool on 32bits, use the simplest refactoring, return an int after calling lastinsertid

I made a lib last weekend for integer types juggling BTW

Feel free to use it

https://dev.to/ccoveille/go-safecast-safe-number-conversion-in-go-4c2l